src/util.ts

```diff
@@ -144,4 +145,31 @@ export function parseAzureResourceId(resourceId: string): { [key: string]: strin
     }
 
     return result;
-}
\ No newline at end of file
+}
+
+export async function showWorkspaceFoldersQuickPick(placeHolderString: string): Promise<vscode.WorkspaceFolder> {
```

Comment:
I don't have strong preferences in regards to your description of the PR, but whatever logic you decide on should be in this function (Rather than the current implementation where every caller of showWorkspaceFoldersQuickPick has to check if workspaceFolders is defined and throw a very similar error message)